accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/authorizableinstaller/impl/AuthInstallerUserManagerPrefetchingImpl.java

```diff
@@ -91,7 +93,15 @@ public AuthInstallerUserManagerPrefetchingImpl(UserManager delegate, final Value
                 Group memberOfGroup = declaredMemberOf.next();
                 String memberOfGroupId = memberOfGroup.getID();
                 isMemberOfByAuthorizableId.get(authId).add(memberOfGroupId);
-                nonRegularUserMembersByAuthorizableId.get(memberOfGroupId).add(authId);
+                final Set<String> nonRegularUserMembers;
+                if (!nonRegularUserMembersByAuthorizableId.containsKey(memberOfGroupId)) {
```

Comment:
this condition is very problematic, because the authorizable installer relies on having a full/complete result of the methods `getDeclaredIsMemberOf()` und `getDeclaredMembersWithoutRegularUsers()` - if the query from the `QueryBuilder` above is not 100% reliable, we should switch to traversal of /home/groups and /home/users/system (having these two roots does not require to traverse over all users fortunately). The traversal could be fallback or default (if performance is similar to the query)

Reply:
Not the if condition is problematic but the unreliable search. I am also fine with throwing another exception to make it more obvious that the index is corrupt/incomplete in this case, but this cannot always be detected reliably. I hope we can agree on the fact that running into an NPE is a situation we should prevent....

Reply:
let's use https://github.com/Netcentric/accesscontroltool/pull/617/files instead then